### ISSUE TITLE
[Bug]: Image-based normalized envrionements failed to load

### ISSUE THREAD

**Issue by qgallouedec** [DLR-RM/stable-baselines3#1320]
### 🐛 Bug

Loading normalized image-based fails.

### To Reproduce

```sh
python -m rl_zoo3.train -env MiniGrid-FourRooms-v0 --algo ppo --gym-packages gym_minigrid -n 10000
python -m rl_zoo3.push_to_hub --env MiniGrid-FourRooms-v0 --algo ppo -f logs -orga qgallouedec
```

### Relevant log output / Error message

```shell
python -m rl_zoo3.push_to_hub --env MiniGrid-FourRooms-v0 --algo ppo -f logs -orga qgallouedec
Loading latest experiment, id=2
Loading logs/ppo/MiniGrid-FourRooms-v0_2/MiniGrid-FourRooms-v0.zip
Loading running average
with params: {'norm_obs': True, 'norm_reward': False}
Wrapping the env in a VecTransposeImage.
Traceback (most recent call last):
  File "/home/qgallouedec/miniconda3/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/qgallouedec/miniconda3/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/qgallouedec/rl-baselines3-zoo/rl_zoo3/push_to_hub.py", line 383, in <module>
    model = ALGOS[algo].load(model_path, env=eval_env, custom_objects=custom_objects, device=args.device, **kwargs)
  File "/home/qgallouedec/stable-baselines3/stable_baselines3/common/base_class.py", line 689, in load
    check_for_correct_spaces(env, data["observation_space"], data["action_space"])
  File "/home/qgallouedec/stable-baselines3/stable_baselines3/common/utils.py", line 228, in check_for_correct_spaces
    raise ValueError(f"Observation spaces do not match: {observation_space} != {env.observation_space}")
ValueError: Observation spaces do not match: Dict(image:Box([[[-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]]

 [[-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]]

 [[-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]]

 [[-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]]

 [[-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]]

 [[-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]]

 [[-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]
  [-10. -10. -10.]]], [[[10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]]

 [[10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]]

 [[10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]]

 [[10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]]

 [[10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]]

 [[10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]]

 [[10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]
  [10. 10. 10.]]], (7, 7, 3), float32)) != Dict(image:Box([[[0 0 0 0 0 0 0]
  [0 0 0 0 0 0 0]
  [0 0 0 0 0 0 0]
  [0 0 0 0 0 0 0]
  [0 0 0 0 0 0 0]
  [0 0 0 0 0 0 0]
  [0 0 0 0 0 0 0]]

 [[0 0 0 0 0 0 0]
  [0 0 0 0 0 0 0]
  [0 0 0 0 0 0 0]
  [0 0 0 0 0 0 0]
  [0 0 0 0 0 0 0]
  [0 0 0 0 0 0 0]
  [0 0 0 0 0 0 0]]

 [[0 0 0 0 0 0 0]
  [0 0 0 0 0 0 0]
  [0 0 0 0 0 0 0]
  [0 0 0 0 0 0 0]
  [0 0 0 0 0 0 0]
  [0 0 0 0 0 0 0]
  [0 0 0 0 0 0 0]]], [[[255 255 255 255 255 255 255]
  [255 255 255 255 255 255 255]
  [255 255 255 255 255 255 255]
  [255 255 255 255 255 255 255]
  [255 255 255 255 255 255 255]
  [255 255 255 255 255 255 255]
  [255 255 255 255 255 255 255]]

 [[255 255 255 255 255 255 255]
  [255 255 255 255 255 255 255]
  [255 255 255 255 255 255 255]
  [255 255 255 255 255 255 255]
  [255 255 255 255 255 255 255]
  [255 255 255 255 255 255 255]
  [255 255 255 255 255 255 255]]

 [[255 255 255 255 255 255 255]
  [255 255 255 255 255 255 255]
  [255 255 255 255 255 255 255]
  [255 255 255 255 255 255 255]
  [255 255 255 255 255 255 255]
  [255 255 255 255 255 255 255]
  [255 255 255 255 255 255 255]]], (3, 7, 7), uint8))
```

### System Info

- OS: Linux-5.15.0-58-generic-x86_64-with-glibc2.35 # 64-Ubuntu SMP Thu Jan 5 11:43:13 UTC 2023
- Python: 3.9.12
- Stable-Baselines3: 1.8.0a4
- PyTorch: 1.11.0+cpu
- GPU Enabled: False
- Numpy: 1.24.1
- Gym: 0.21.0

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] I have provided a minimal working example to reproduce the bug
- [X] I've used the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by araffin** [comment]
> ValueError: Observation spaces do not match

oh I see, one is using the updated space from vecnormalize, where the other is not.


I also think we should allow the user to by pass the check, because an env might still be valid (correct shape) despite slightly different bounds.

**Comment by qgallouedec** [comment]
## Actually it's a bit more complex than it looks.

In gym-minigrid, the observation is a **fake image**: from the [doc](https://minigrid.farama.org/environments/minigrid/UnlockEnv/)

> Each tile is encoded as a 3 dimensional tuple: (OBJECT_IDX, COLOR_IDX, STATE)

And to make it even more complex, it is encapsulated in a Dict. So we have `env.observation_space == Dict("image": Box(0, 255, (map_width, map_height, 3), np.uint8)`.

So it is not an image, but it is detected as such in sb3. Here is what happens:

First, we pass `normalize=True` (see https://github.com/DLR-RM/rl-baselines3-zoo/blob/33eba22eb36128412a5b22b57a7a10bfe71e6278/hyperparams/ppo.yml#L292), which wraps the environment, and changes the observation space to `env.observation_space == Dict("image": Box(-10, 10, (map_width, map_height, 3), np.float32)`.

Then, since the policy is `MultiInput` (because remember, the observation is actually a `Dict`), the nature of the feature extractor is set automatically. At this point, the new observation space does not fulfill the condition to be an image (because of its dtype). Thus, the feature_extractor chosen is Flatten.
(If we had an environment without Dict, and we had passed policy=CnnPolicy, the feature extractor would have raised an error).

And this is how two "errors" cancel each other out, and how the learning process goes on as desired, i.e. by not considering the observation as an image.

In short, in the case of gym-minigrid, I think you should use a FlattenWrapper to avoid unexpected behavior.

## For the rest,

the problem remains, but only in the specific case of observations already normalized (and therefore channel-first). In this case, the troublesome code is:

```python
def env_func():
    return FakeImageEnv(channel_first=True)

venv = VecNormalize(DummyVecEnv([env_func]))
model = DQN("CnnPolicy", venv, policy_kwargs=dict(normalize_images=False))
model.save("test_save.zip")
venv.save("vecnormalize.pkl")

venv = DummyVecEnv([env_func])
venv = VecNormalize.load("vecnormalize.pkl", venv)
DQN.load("test_save.zip", env=venv)
```
